src/config/cmd.rs

```diff
@@ -340,8 +343,11 @@ mod tests {
             let config_source = create_config_source(sources.as_ref(), transforms.as_ref(), sinks.as_ref());
 
             // Shuffle the ordering of components which shuffles the order in
-            // which items appear in the final config
-            let mut rng = thread_rng();
+            // which items appear in the TOML config
+            let mut rng = StdRng::from_seed([
+                    3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3, 2, 3, 8, 4, 6, 2, 6, 4, 3, 3,
+                    8, 3, 2, 7, 9, 5,
+                ]);
```

Comment:
Instead of a manual seed here you might consider passing this in as a generated parameter. It'll be tidier, for one, but it will also let the property test mechanism explore different sort orders for the same components.

Reply:
Added in [11facf7](https://github.com/vectordotdev/vector/pull/12713/commits/11facf70a40906bf30549b89cb8b0026b4c26634). There's gotta be something poetic(?) about randomly generating a thing whose purpose is to control randomness.